exporters/otlp/otlplog/otlploghttp/client_test.go

```diff
@@ -700,6 +700,33 @@ func TestConfig(t *testing.T) {
 		assert.Len(t, rCh, 0, "failed HTTP responses did not occur")
 	})
 
+	t.Run("WithRetryAndExporterErr", func(t *testing.T) {
+		exporterErr := errors.New("rpc error: code = Unavailable desc = service.name not found in resource attributes")
+		rCh := make(chan exportResult, 1)
+		rCh <- exportResult{Err: &httpResponseError{
+			Status: http.StatusTooManyRequests,
+			Err:    exporterErr,
+		}}
+		exp, coll := factoryFunc("", rCh, WithRetry(RetryConfig{
+			Enabled: false,
+		}))
+		ctx := context.Background()
+		t.Cleanup(func() { require.NoError(t, coll.Shutdown(ctx)) })
+		// Push this after Shutdown so the HTTP server doesn't hang.
+		t.Cleanup(func() { close(rCh) })
+		t.Cleanup(func() { require.NoError(t, exp.Shutdown(ctx)) })
+		err := exp.Export(ctx, make([]log.Record, 1))
+		assert.Error(t, err, exporterErr)
```

Comment:
This is testing `err` is not nil. It is using `exporterErr` as a message: https://pkg.go.dev/github.com/stretchr/testify/assert#Error

```suggestion
		assert.ErrorIs(t, err, exporterErr)
```

Reply:
Changed it to ErrorContains, as ErrorIs will not work here, sample link for explanation 
https://go.dev/play/p/5wzdTmyy1pD